packages/mosip-api/src/mosip-api.ts

```diff
@@ -8,32 +10,265 @@ export class MOSIPError extends Error {
   }
 }
 
+export async function getMosipAuthToken() {
+  console.log('env.MOSIP_AUTH_URL: ', env.MOSIP_AUTH_URL, env.MOSIP_AUTH_CLIENT_ID, env.MOSIP_AUTH_CLIENT_SECRET)
+  const response = await fetch(env.MOSIP_AUTH_URL, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json"
+      // cookie: `Authorization=${authToken}; OpenCRVSToken=${token};`,
+    },
+    body: `
+    {
+      "id": "string",
+      "version": "string",
+      "requesttime": "2025-03-11T02:53:07.206Z",
+      "metadata": {},
+        "request": {
+            "clientId": "${env.MOSIP_AUTH_CLIENT_ID}",
+            "secretKey": "${env.MOSIP_AUTH_CLIENT_SECRET}",
+            "appId": "admin"
+        }
+    }
+    `
+  });
+
+  if (!response.ok) {
+    throw new MOSIPError(
+      `Failed getting MOSIP auth token. Response: ${response.status
+      }, response: ${await response.text()}`,
+    );
+  }
+  const responseJson = await response.json();
+  // Get the 'Set-Cookie' header from the response
+  const cookie: string | null = response.headers.get('Set-Cookie');
+
+  if (!cookie || cookie === null) {
+    throw new MOSIPError(
+      `Failed getting MOSIP auth token. Response: ${response.status
+      }, response: ${await response.text()}`,
+    );
+  }
+  console.log('responseJson', responseJson);
+  // const token = (await response.json()) as { access_token: string };
+  // Split the string by ';' to separate the cookie parts
+  const cookieParts = cookie.split(';');
+
+  // The first part will be the Authorization token
+  const authorizationPart = cookieParts[0];
+
+  // Extract the token by splitting on '='
+  const token = authorizationPart.split('=')[1];
+  console.log('token: ', token)
+  return token;
+}
+
 export const postBirthRecord = async ({
   event,
   token,
+  request
 }: {
   event: {
     id: string;
     trackingId: string;
   };
   token: string;
+  request: OpenCRVSRequest;
 }) => {
-  const response = await fetch(env.MOSIP_BIRTH_WEBHOOK_URL, {
-    method: "POST",
-    body: JSON.stringify({ event, token }),
+
+  const composition = getComposition(request.body);
+  
+    const child = findEntry(
+      "child-details",
```

Comment:
Recommending you to add **Prettier: Code formatter** to VSCode and enable [Format on Save](https://stackoverflow.com/a/54665086). This way the code is always formatted the same for every developer.